dftimewolf/lib/state.py

```diff
@@ -0,0 +1,73 @@
+"""This class maintains the internal dfTimewolf state.
```

Comment:
Seeing that this new state machine introduces complexity, I opt for some design/documentation.

Reply:
Sounds good. Where do we put it?

Reply:
I suggest the github wiki, under developer docs.